plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TrinoHiveCatalog.java

```diff
@@ -215,7 +219,25 @@ public boolean dropNamespace(ConnectorSession session, String namespace)
                 !listViews(session, Optional.of(namespace)).isEmpty()) {
             throw new TrinoException(SCHEMA_NOT_EMPTY, "Schema not empty: " + namespace);
         }
-        metastore.dropDatabase(new HiveIdentity(session), namespace);
+
+        Optional<Path> location = metastore.getDatabase(namespace)
+                .orElseThrow(() -> new SchemaNotFoundException(namespace))
+                .getLocation()
+                .map(Path::new);
+
+        boolean deleteData = location.map(path -> {
```

Comment:
There seems to exist the same check routine in two different places:
- `TrinoHiveCatalog.java`
- `SemiTransactionalHiveMetastore.java`

https://github.com/trinodb/trino/pull/9767/files#diff-4437180ac9a56604967e9ae5f4d4aafdc3804d93faa2c0011eca4fbafb83062aR382-R392

Is there any chance to reuse the code?

Reply:
We could probably put the key shared logic here into a utility method (maybe in `HiveUtil`), something like `locationContainsFiles(Path, Session)`.

Other than that, I don't think there are great options to de-duplicate this code.

Reply:
Let's leave it as is for now, and revisit later, if we find a nice way to package that